### PR TITLE
adding deregister action to batch-definition

### PR DESCRIPTION
**Pull request by joshuaroot**
Per #2268
- adding action deregister batch-definition object

### CODE REVIEW

File: c7n/resources/batch.py [hunk 1]
```diff
@@ -152,3 +154,42 @@ def process(self, resources):
...
+    schema = type_schema('deregister')
+    permissions = ('batch:DeregisterJobDefinition',)
+
+    def deregister_definition(self, r):
+        try:
+            self.client.deregister_job_definition(
+                jobDefinition='%s:%s' % (r['jobDefinitionName'],
+                                         r['revision']))
+        except ClientError as e:
+            if e.response['Error']['Code'] in ('ClientException',
```
[the remark this answers is not in the document]
Comment (joshuaroot) [reply c7n/resources/batch.py]: The definitions of these exceptions are just as generic on the AWS docs page as well...

https://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_DeregisterTaskDefinition.html#API_DeregisterTaskDefinition_Errors

[the remark this answers is not in the document]
Comment (kapilt) [reply c7n/resources/batch.py]: we shouldn't catch them then, unless we can more specific about what specifically we're catching. ie. per zen of python errors should never pass unless explicitly silenced.

[the remark this answers is not in the document]
Comment (joshuaroot) [reply c7n/resources/batch.py]: Was able to get 1 error by forcing an incorrect value in the code but I don't anticipate this actually being seen as the code formats the `jobDefinitionName:jobRevision` automatically.

`An error occurred (ClientException) when calling the DeregisterJobDefinition operation: Provided name c7n:3A is not supported pattern[a-zA-Z_0-9-]{1,128}`

**Comment by kapilt** [comment]
needs a merge of trunk to resolve conflicts

**Pull request merged**